tests/zquantum/core/estimation/estimation_test.py

```diff
@@ -696,3 +696,19 @@ def test_calculate_exact_expectation_values_fails_with_non_simulator(
         backend = MockQuantumBackend()
         with pytest.raises(AttributeError):
             _ = calculate_exact_expectation_values(backend, estimation_tasks)
+
+    def test_no_measured_estimation_tasks_provided(simulator):
+        estimation_tasks = [
+            EstimationTask(IsingOperator("Z0"), Circuit([H(0)]), 0),
+        ]
+
+        target = [
+            ExpectationValues(np.array([0.0]), [np.array([[0.0]])], [np.array([0.0])])
+        ]
+
+        expectation_values_list = estimate_expectation_values_by_averaging(
+            simulator, estimation_tasks
+        )
+
+        # test that all attributes are the same
+        assert expectation_values_list[0].__dict__ == target[0].__dict__
```

Comment:
We can't just compare two `ExpectationValues` objects? Maybe we should be able to do that?

Reply:
[request] Seems like a good idea to me to also check that the length of `expectation_values_list` is 1?

Reply:
Added a method of comparing ExpectationValues and assertion that list has length 1.

We might want to make the comparison method a little more sophisticated than what I have written.